### PR TITLE
Remove team adapters

### PR DESCRIPTION
**Pull request by zoldar**
### Changes

This PR completely removes `Teams` read adapters replacing them with direct references to team-centric implementation.

### Tests
- [x] Automated tests have been added

### CODE REVIEW

File: lib/plausible/billing/feature.ex [hunk 1]
```diff
@@ -210,7 +216,38 @@ defmodule Plausible.Billing.Feature.StatsAPI do
...
+      {unlimited_trial?, subscription?}
+
+      owner = team && Plausible.Repo.preload(team, :owner).owner
+
+      pre_business_tier_account? =
+        not is_nil(owner) and
+          NaiveDateTime.before?(owner.inserted_at, Plausible.Billing.Plans.business_tier_launch())
+
+      cond do
+        # FIXME: this is unreachable code - unlimited_trial? means that basically team is nil
```
Comment (zoldar) [inline lib/plausible/billing/feature.ex:236]: \cc @aerosol

File: test/plausible/billing/feature_test.exs [hunk 2]
```diff
@@ -6,96 +6,100 @@ defmodule Plausible.Billing.FeatureTest do
...
     assert {:error, :upgrade_required} ==
-             Plausible.Billing.Feature.StatsAPI.check_availability(user)
+             Plausible.Billing.Feature.StatsAPI.check_availability(team)
   end
 
+  @tag :skip
   test "Plausible.Billing.Feature.StatsAPI.check_availability/2 returns :ok when user trial hasn't started and was created before the business tier launch" do
-    user = new_user(inserted_at: ~N[2020-01-01T00:00:00], trial_expiry_date: nil)
-    assert :ok == Plausible.Billing.Feature.StatsAPI.check_availability(user)
+    # FIXME can't trigger this condition
```
Comment (zoldar) [inline test/plausible/billing/feature_test.exs:70]: Ditto \cc @aerosol

File: lib/plausible/billing/feature.ex (hunk 1 above)
Comment (aerosol) [reply lib/plausible/billing/feature.ex:236]: how about `unlimited_trial? = team and is_nil(team.trial_expiry_date) and is_nil(team.subscription)`.

I'll ask Marko about this.

**Review by aerosol — APPROVED** [review]
:rocket:

**Pull request merged**